components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/UserResource.java

```diff
@@ -145,6 +147,12 @@ public Response deleteUser(@PathParam(SCIMProviderConstants.ID) String id,
             // obtain the user store manager
             UserManager userManager = IdentitySCIMManager.getInstance().getUserManager();
 
+            String superAdminID = AdminAttributeUtil.getSuperAdminID();
+            String loggedInUser = SCIMCommonUtils.getLoggedInUserID();
+            if ((superAdminID.equals(id)) && (!loggedInUser.equals(id))) {
```

Comment:
what if logged in user has required role to perform delete action? 
why this delete operation restricted to super admin user?

Reply:
The operation to delete the SuperAdmin is only limited to the SuperAdmin themselves. No other user should be allowed to perform this action.